src/Stormpath/Client.php

```diff
@@ -192,5 +197,13 @@ public static function tearDown()
         static::$instance = NULL;
     }
 
+    private function resolveSigner()
+    {
+        $signer = "\\Stormpath\\Http\\Authc\\" . self::$authenticationScheme . "Signer";
+        if(!class_exists($signer))
+            $signer = "\\Stormpath\\Http\\Authc\\" . Stormpath::AUTHENTICATION_SCHEME_SAUTHC1 . "Signer";
+        return $signer;
```

Comment:
If the user explicitly specified an unknown authentication scheme we should throw an `InvalidArgumentException` instead of ignoring what was supplied. Also, we should move this logic to the `build` method of the `ClientBuilder`. 

The `ClientBuilder` might have to use reflection to get the `requestExecutor`, but that should be OK since this is only to create a `Client` instance.